experimental/relativetime/src/provider.rs

```diff
@@ -0,0 +1,124 @@
+    pub display_name: Cow<'data, str>,
+    /// Mapping for relative times with unique names.
+    /// Example.
+    /// In English, "-1" corresponds to "yesterday", "1" corresponds to "tomorrow".
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub relatives: ZeroMap<'data, i8, str>,
+    /// How to display times in the past.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub past: PluralRulesCategoryMapping<'data>,
+    /// How to display times in the future.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub future: PluralRulesCategoryMapping<'data>,
+}
+
+/// Display specification for relative times, split over potential plural patterns.
+#[derive(Debug, Clone, Default, PartialEq, yoke::Yokeable, zerofrom::ZeroFrom)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[cfg_attr(
+    feature = "datagen", 
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_relativetime::provider)
+)]
+#[yoke(prove_covariance_manually)]
+pub struct PluralRulesCategoryMapping<'data> {
+    /// Mapping for PluralCategory::Zero or `None` if not present.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub zero: Option<SingularSubPattern<'data>>,
+    /// Mapping for PluralCategory::One or `None` if not present.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub one: Option<SingularSubPattern<'data>>,
+    /// Mapping for PluralCategory::Two or `None` if not present.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub two: Option<SingularSubPattern<'data>>,
+    /// Mapping for PluralCategory::Few or `None` if not present.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub few: Option<SingularSubPattern<'data>>,
+    /// Mapping for PluralCategory::Many or `None` if not present.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub many: Option<SingularSubPattern<'data>>,
+    /// Mapping for PluralCategory::Other or `None` if not present.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub other: Option<SingularSubPattern<'data>>,
+}
+
+/// Singular substitution pattern string.
+#[derive(Debug, Clone, Default, PartialEq, yoke::Yokeable, zerofrom::ZeroFrom)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[cfg_attr(
+    feature = "datagen", 
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_relativetime::provider)
+)]
+pub struct SingularSubPattern<'data> {
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    /// The underlying pattern
+    pub pattern: Cow<'data, str>,
+    /// Optional index which is absent if the pattern does not have substitution
+    pub index: Option<u8>,
+}
+
+impl<'data> SingularSubPattern<'data> {
+    /// Construct a singular sub pattern from a pattern
+    pub fn try_from_str(value: &str) -> Result<Self, DataError> {
+        let index = value.find("{0}").map(|index| index as u8);
+        Ok(Self {
+            pattern: Cow::Owned(value.to_string()),
```

Comment:
Suggestion: remove the `{0}` from the string and rely only on the index.

Alternatively, just keep the placeholder and drop the index, or perhaps replace the placeholder with some other character that won't show up in real localized strings, like one of the ASCII control codes. This is a really cheap runtime parse that I'd be comfortable with.

Reply:
Removing the `{0}` placeholder and using index with 255u8 denoting no placeholder.